WMIAdventure/backend/WMIAdventure_backend/battle/migrations/insert_test_players_data.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 3.2.1 on 2021-06-07 20:48
+import random
+
+from django.db import migrations
+
+from IngameUsers.models import UserProfile, Semester, Deck
+from cards.models import Card, CardInfo, CardLevel, CardEffect, CardLevelEffects
+from users.models import User
+
+
+def create_test_cards():
+    """
+    Creates test cards if there are less than 5 cards in the database.
+    At least 5 cards are required to create test users.
+
+    :return: None
+    """
+    cards_count = len(Card.objects.all())
+    while cards_count < 5:
+        create_test_card(cards_count)
+        cards_count += 1
+
+
+def create_test_card(cards_count: int):
+    """
+    Creates test card.
+
+    :param cards_count: How many cards are in the database. Will be used to generate unique test card name.
+    :return: None
+    """
+    test_card_info = CardInfo.objects.create(name=f"test_card{cards_count}", tooltip="test_card")
```

Comment:
Powinienem sprawdzać, czy nie istnieje już przypadkiem karta o nazwie "test_card4"? Jeśli istnieje to usunąć taką i stworzyć jeszcze raz?

Reply:
Nie wiem, czy to jest dobry pomysł. Słabo będzie pokazywać karty o takich nazwach, nie lepiej wykorzystać te, które Wirus już zrobił?
Zrobiłbym taką metodę, która pobiera kartę z bazy, a jeżeli takiej nie ma to tworzy test_card{random_number}.

Reply:
Zobacz funkcję create_test_cards, tworzy ona takie testowe karty, tylko wtedy, jak jest mniej niż 5 kart w bazie danych. Tam gdzie Wirus już stworzył prawdziwe karty na azurze, czy produkcji to ta metoda nie stworzy żadnych testowych kart z takimi nazwami.